### ISSUE TITLE
CARTO API deprecated

### ISSUE THREAD

**Issue by gassc** [CodeForPittsburgh/fishfryform#14]
CARTO's pricing model for their API access changed; consequently we won't be able to utilize CARTO as a database for the Fish Fry data. This breaks pretty much everything here. However...

We can migrate the database over to a PostgreSQL DB where the web application is currently living (on a pythonanywhere account), and implement some off-the-shelf Python Flask extensions to handle all the CRUD operations and UI stuff (that we otherwise had to custom build to make this work with CARTO).

**Comment by mhowe0422** [comment]
Can you backup the DB as a tar file and post in GitHub

**Comment by gassc** [comment]
@mhowe0422 see https://github.com/CodeForPittsburgh/fishfryform/commit/aa3d58f7222d47d5dfe09b08fab902b6cc977de8 for database dump

**Comment by mhowe0422** [comment]
Hi Christian:


I updated my local git hub copies with your changes for the Fish Fry. (map/form)

I can't open the gpkg with any software I have. qgis looked at me funny!

The file looks like is was created with sqlite?


Looking at the csv files, there are columns that we might not want transferred and we should generate our own UID


I'm thinking that if we just had the map populated from editable static data. This is sort of what I have

on Git Hub for the daily police blotter map. (google map and xml data)

We could then work on suitable data entry forms.


We are in short supply of coding talent. Will we have a project management resource issue trying to launch both the Unlandmark and the 2018 Fish Fry?


I keep updating my local unlandmark tables based on on-going issue solutions.


So, my vote is for me to continue with Unlandmark and you forge ahead with the Fish Fry.


Mark


________________________________

**Comment by gassc** [comment]
Hey Mark,

That .gpkg is an export from CARTO. It will open for sure with QGIS 2.18 (as a vector file) – I tested it before publishing to GitHub. It may very well be SQLite (or SpatiaLite).

I figured all that extraneous ID stuff can go – some of that is from CARTO, and some I put there to workaround some limitations of the CARTO SQL API.

Going forward, I’d rather not force the map to depend on static data files (maybe as a fallback). The project originated with a Google Sheet right? No going back now!

For the data-entry/data-management side of things, given that what I have now will break at midnight tonight: I’m going to implement Flask-Admin with SQLAlchemy, pointed as a PostgreSQL/PostGIS database (someone else's example here<http://examples.flask-admin.org/geo/admin/point/>), and rewrite the logic behind the Fish Fry API method(s) accordingly to generate the geojson. I’ve used it with success on a few of my projects, so I have working, tested, production code and a deployment pattern worked out for reference. It’s a option that will require minimal lift on my part, will address the form validation side of things, and give Hollen the interface she needs.

I’m planning to let Hollen keep doing her thing with curating public submissions from her Facebook page using this tool. This year she’ll have the advantage of having all of last year’s locations queued up, with most of the work encompassing verification for existing fish frys.

Adding Unlandmarks on top of this is going to be a tough lift for me, to be honest; and yes, we are short on people. I don’t imagine that’s going to improve anytime soon. Let’s plan to talk about that at the next work night.

Christian
